### PR TITLE
[CLOUDCI-467] Fix update state history

### PR DESCRIPTION
**Pull request by rinckm**
### Description

Fix update state history

### Submitter checklist

- [ ] Change has been tested (on a back-end cluster)
- [ ] (If applicable) Jira backlog item ID added to the PR title and the [changelog.yaml] entry
- [x] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [x] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- [ ] Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- [ ] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [ ] Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There is at least 1 approval for the pull request and no outstanding requests for change
- [ ] All voter checks have passed
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] The Pull Request title is understandable and reflects the changes well
- [ ] The Pull Request description is understandable and well documented
- [ ] [changelog.yaml] entry for this Pull Request has been added
    - [ ] Changelog entry contains all required information
    - [ ] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: pkg/k8s/pipelineRun.go [hunk 1]
```diff
@@ -128,46 +128,55 @@ func (r *pipelineRun) UpdateState(state api.State) (*api.StateItem, error) {
...
 		r.apiObj.Status.State = state
+		return nil
 	})
+
+	if err != nil {
+		return nil, err
+	}
+	oldState.State = api.StateNew
+	oldState.StartedAt = r.apiObj.ObjectMeta.CreationTimestamp
+	return &oldState, nil
```
Comment (anbrsap) [inline pkg/k8s/pipelineRun.go:164]: This is wrong in case the initial state was set already.

Comment (anbrsap) [inline pkg/k8s/pipelineRun.go:164]: In variable names the term "state is used both for plain state names and state details. This is confusing.

I would rename `{old,current,new]State` to `{old,current,new}StateDetails`.

Comment (rinckm) [reply pkg/k8s/pipelineRun.go]: Done

Comment (rinckm) [reply pkg/k8s/pipelineRun.go]: Take result from history now and improved tests.

**Review by anbrsap — APPROVED** [review]

**Pull request merged**